src/PlaywrightSharp.Abstractions/Extensions.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Threading.Tasks;
+
+namespace PlaywrightSharp
+{
+    /// <summary>
+    /// <see cref="IJSHandle"/> and <see cref="IElementHandle"/> Extensions.
+    /// </summary>
+    public static class Extensions
+    {
+        /// <summary>
+        /// Runs <paramref name="pageFunction"/> within the frame and passes it the outcome of <paramref name="elementHandleTask"/> as the first argument.
+        /// </summary>
+        /// <param name="elementHandleTask">A task that returns an <see cref="IElementHandle"/> that will be used as the first argument in <paramref name="pageFunction"/>.</param>
+        /// <param name="pageFunction">Function to be evaluated in browser context.</param>
+        /// <param name="args">Arguments to pass to <c>pageFunction</c>.</param>
+        /// <returns>Task.</returns>
+        public static async Task EvaluateAsync(this Task<IElementHandle> elementHandleTask, string pageFunction, params object[] args)

```

Comment:
We are going to get closer to the Playwright API here. We already have a `QuerySelectorEvaluateAsync` method on IPage and IFrame.

Reply:
Let's move back the SelectionException to the PlaywrightSharp library.